Sources/StreamChatUI/ChatMessageList/Attachments/ChatMessageVoiceRecordingAttachmentListView+ItemView.swift

```diff
@@ -0,0 +1,342 @@
+                isRecording: false,
+                isPlaying: false,
+                duration: content.extraData?.duration ?? 0,
+                currentTime: TimeInterval(waveformView.slider.value),
+                waveform: content.extraData?.waveform ?? []
+            )
+        }
+
+        // MARK: - View updates
+
+        open func updatePlayPauseButton(
+            for state: AudioPlaybackState
+        ) {
+            playPauseButton.setImage(appearance.images.playFill, for: .normal)
+            playPauseButton.isSelected = state == .playing
+            switch state {
+            case .notLoaded, .loading:
+                playPauseButton.setImage(nil, for: .selected)
+            case .paused, .playing, .stopped:
+                playPauseButton.setImage(appearance.images.pauseFill, for: .selected)
+            default:
+                break
+            }
+        }
+
+        open func updatePlaybackLoadingIndicator(
+            for state: AudioPlaybackState
+        ) {
+            switch state {
+            case .loading:
+                playbackLoadingIndicator.isHidden = false
+                playbackLoadingIndicator.startRotation()
+            default:
+                playbackLoadingIndicator.isHidden = true
+                playbackLoadingIndicator.stopRotating()
+            }
+        }
+
+        open func updateDurationLabel(
+            for state: AudioPlaybackState,
+            duration: TimeInterval,
+            currentTime: TimeInterval
+        ) {
+            switch state {
+            case .notLoaded, .loading:
+                durationLabel.isHidden = false
+                durationLabel.text = appearance.formatters.videoDuration.format(duration)
+            case .stopped:
+                durationLabel.isHidden = false
+                durationLabel.text = appearance.formatters.videoDuration.format(duration)
+            case .paused, .playing, .stopped:
+                durationLabel.isHidden = false
+                durationLabel.text = appearance.formatters.videoDuration.format(min(currentTime, duration))
+            default:
+                break
+            }
+        }
+
+        open func updateWaveformView(
+            for state: AudioPlaybackState,
+            duration: TimeInterval,
+            currentTime: TimeInterval
+        ) {
+            waveformView.content = .init(
+                isRecording: false,
+                isPlaying: state == .paused || state == .playing,
+                duration: duration,
+                currentTime: currentTime,
+                waveform: waveformView.content.waveform
+            )
+        }
+
+        open func updateFileIconImageView(
+            for state: AudioPlaybackState
+        ) {
+            switch state {
+            case .notLoaded, .loading, .stopped:
+                fileIconImageView.isHidden = false
+            case .paused, .playing:
+                fileIconImageView.isHidden = true
+            default:
```

Comment:
In general, I see a huge usage of `default`, which I consider quite "dangerous" or that can produce some issues when adding new cases. Is this done for any particular reason?

Reply:
This is a consequence of using structs instead of enums for this kind of objects. Without the `default` there will be a compiler error.

Reply:
Maybe we can add an assertion for the cases where adding a new enum could be problematic 🤔 WDYT?

Reply:
Missed the point on them being structs, yeah. A bit of a pain, but understandable